lib/Common/Memory/HeapAllocator.h

```diff
@@ -93,6 +93,57 @@ struct HeapAllocatorData
 
 struct HeapAllocator
 {
+    template<typename T>
+    struct AutoFree
+    {
+    private:
+        T * obj = nullptr;
+    public:
+        ~AutoFree()
+        {
+            HeapDelete(obj);
+        }
+        void Replace(T* obj)
```

Comment:
nit: maybe assert this->obj is null to start/rename this to Set instead of Replace